README.md

```diff
@@ -1,6 +1,306 @@
+
+- Auth:  The S3 api implementation (Gluster-Swift) does not enforce any authentication / authorization.
+Each new bucket, regardless of the *Namespace* of its *ServiceInstance*, is accessible and mutable by anyone with the coordinates of the S3 server.
+
+- Flat Bucket Hierarchy:  Gluster-Swift's S3 implementation allows for nested buckets.
+However, we have written the CNS Object Broker utilizing the minio-go S3 client, which has no concept of nested buckets.
+This results in an artificially flat bucket hierarchy.
+
+## Installation
+
+### Dependencies
+- [Google SDK](https://cloud.google.com/sdk/) (`gcloud`): To access GCP via cli.
+- [Kubernetes](https://github.com/kubernetes/kubernetes): to run local K8s cluster
+- [Gluster-Kubernetes Deploy Script](https://github.com/copejon/gk-cluster-deploy): to deploy GCP K8s cluster with Gluster-Kubernetes Object Storage and CNS Object Broker
+- Kubernetes-Incubator [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog): to provide Service-Catalog helm chart
+- Kubernetes [Helm](https://github.com/kubernetes/helm#install): to install Service-Catalog chart
+
+### Assumptions
+- Gk-cluster-deploy was written and tested on Fedora 25 and RHEL 7.
+- A [Google Cloud Platform](https://cloud.google.com/compute/) account.
+
+### Topology
+There are two primary systems that make up this demonstration
+They are the Broker and its colocated CNS Object store.
+It should be noted that the Broker can be implemented to run anywhere
+They fill the role of our micro-service provider
+It is only for the purpose of this demo that we decided deploy the Broker and the CNS Cluster in the same location
+
+
+A second system will be the locally running Kubernetes cluster on which with Service-Catalog is deployed
+This cluster will be our micro-server client
+Please refer to the [command flow diagram](docs/diagram/control-diag.md) for a more in depth look at these systems.
+
+## Setup
+### Step 0: Preparing environment
+- Clone [Kubernetes](https://github.com/kubernetes)
+- Clone [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog)
+- Install [Google SDK](https://cloud.google.com/sdk/) and add `gcloud` to `PATH`
+You must have an active Google Cloud Platform account.
+- Configure `gcloud` user, zone, and project
+These are detected by the deployment script and required for setup.
+
+
+`# gcloud config set account <user account>`
+
+`# gcloud config set project <project>`
+
+`# gcloud config set compute/zone <zone>`
+
+
+### Step 1: Deploy Gluster-Kubernetes cluster in Google Compute Engine (GCE)
+This step sets up the **External Service Provider** portion of our topology (see [diagram](docs/diagram/control-diag.md)).
```

Comment:
_sets up the **External Service Provider** portion of..._
Should be be consistent and use **Broker** instead of provider? If not what do you think about **Broker/Provider**?

Reply:
That's a good point.  

Here the _service broker_ functions as a part of the _external service provider._  So the ESP is the sum of the infrastructure, broker, and storage components.